### PR TITLE
Use get_primary_bam utility in macs helper function

### PR DESCRIPTION
**Pull request by jvrakor**
@dblenkus @jkokosar @mstajdohar

### CODE REVIEW

File: resdk/resources/sample.py [hunk 1]
```diff
@@ -19,7 +19,10 @@ def get_reads(self):
 
     def get_bam(self):
         """Return ``bam`` object on the sample."""
-        return self.data.get(type='data:alignment:bam:')
+        try:
+            return self.data.get(type='data:alignment:bam:primary:')
```
Comment (dblenkus) [inline resdk/resources/sample.py]: I don't like that `get_bam` function returns primary bam. You should add `get_primary_bam` function to make it cleaner and it can returns normal bam if primary is missing (you can also add flag to enable/disable this behaviour).

File: docs/CHANGELOG.rst [hunk 2]
```diff
@@ -13,6 +13,8 @@ Changed
 -----
 * Remove ``threads`` parameter from ``cuffquant`` and ``cufnorm``
   helper functions
+* Modify ``get_bam`` utility function to correctly work with 
+  ``bam-split`` processed samples
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Point out that this change is required to work with the latest process version - it is not a bug.

File: resdk/resources/sample.py (hunk 1 above)
Comment (jvrakor) [reply resdk/resources/sample.py]: What about just adding an optional `primary` flag/input to `get_bam` function? Something like:
```def get_bam(self, primary=False):
       if primary:
           try:
               return self.data.get(type='data:alignment:bam:primary:')
           ...
       else:
           return self.data.get(type='data:alignment:bam:')```

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -9,6 +9,15 @@ All notable changes to this project are documented in this file.
 Unreleased
 ==========
 
+Added
+-----
+* Add ``get_primary_bam`` utility function
+
+Changed
+-------
+* Replace `get_bam`` with ``get_primary_bam`` utility function in
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Describe how this affects users, not how you achieve it, e.g. "Use only primary bam files in ``macs`` helper function"

File: resdk/resources/sample.py [hunk 4]
```diff
@@ -21,6 +21,21 @@ def get_bam(self):
...
+
+        If the ``primary bam`` object is not present the function will
+        try to return a ``bam`` object if bam=True.
+
+        """
+        try:
+            return self.data.get(type='data:alignment:bam:primary:')
+        except LookupError:
+            if bam:
+                return self.get_bam()
```
Comment (dblenkus) [inline resdk/resources/sample.py]: I don't like this as ``get_primary_bam`` name implies that primary bam will be returned.

File: resdk/resources/sample.py [hunk 5]
```diff
@@ -21,6 +21,21 @@ def get_bam(self):
         """Return ``bam`` object on the sample."""
         return self.data.get(type='data:alignment:bam:')
 
+    def get_primary_bam(self, fallback_to_bam=False):
+        """Return ``primary bam`` object on the sample.
+
+        If the ``primary bam`` object is not present the function will
+        try to return a ``bam`` object if fallback_to_bam=True.
```
Comment (dblenkus) [inline resdk/resources/sample.py]: Reword this to: 

    If the ``primary bam`` object is not present and ``fallback_to_bam`` is set
    to ``True``, a ``bam`` object will be returned.
